src/agent/src/unix_main.cpp

```diff
@@ -0,0 +1,89 @@
+#include <cmd_ln_parser.hpp>
+#include <logger.hpp>
+#include <process_options.hpp>
+#include <register.hpp>
+
+#include <unix_daemon.hpp>
+
+#include <chrono>
+#include <iostream>
+
+int main(int argc, char* argv[])
+{
+    Logger logger;
+    CommandlineParser cmdParser(argc, argv);
+
+    LogInfo("Starting Wazuh Agent service.");
+
+    try
+    {
+        if (cmdParser.OptionExists("register"))
+        {
+            if (cmdParser.OptionExists("--user") && cmdParser.OptionExists("--password") &&
```

Comment:
I think this code can also be moved to a function (`RegisterAgent`) in `process_options` files.